From ba3b709e08e155a10abb3e88145fb7a4d342a0aa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Tue, 11 Feb 2025 15:42:43 +0100
Subject: [PATCH] cursor-renderer/native: Skip init_hw_cursor_support_for_gpu
 for headless
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

By definition, headless means no HW display output, so initializing HW
cursor support makes no sense.

Fixes hitting the g_warning in tests when there's a GPU device
available, breaking them.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4259>
---
 src/backends/native/meta-cursor-renderer-native.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/backends/native/meta-cursor-renderer-native.c b/src/backends/native/meta-cursor-renderer-native.c
index 7581d9b492..2686e811b4 100644
--- a/src/backends/native/meta-cursor-renderer-native.c
+++ b/src/backends/native/meta-cursor-renderer-native.c
@@ -1350,12 +1350,17 @@ static void
 init_hw_cursor_support_for_gpu (MetaGpuKms *gpu_kms)
 {
   MetaKmsDevice *kms_device = meta_gpu_kms_get_kms_device (gpu_kms);
+  MetaKms *kms = meta_kms_device_get_kms (kms_device);
+  MetaBackend *backend = meta_kms_get_backend (kms);
   MetaCursorRendererNativeGpuData *cursor_renderer_gpu_data;
   const MetaFormatInfo *format_info;
   struct gbm_device *gbm_device;
   uint64_t width, height;
   MetaDrmFormatBuf tmp;
 
+  if (meta_backend_is_headless (backend))
+    return;
+
   cursor_renderer_gpu_data =
     meta_create_cursor_renderer_native_gpu_data (gpu_kms);
 
-- 
2.48.1

